concordium-contracts-common/src/impls.rs

```diff
@@ -163,6 +163,21 @@ impl Deserial for Amount {
     }
 }
 
+impl Serial for ExchangeRate {
+    fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> {
+        out.write_u64(self.numerator())?;
+        out.write_u64(self.denominator())
+    }
+}
+
+impl Deserial for ExchangeRate {
+    fn deserial<R: Read>(source: &mut R) -> ParseResult<Self> {
+        let numerator = source.read_u64()?;
+        let denominator = source.read_u64()?;
+        Ok(ExchangeRate::new_unchecked(numerator, denominator))
```

Comment:
Are we sure we want the unchecked version here? I know that we don't have `gcd` available as is, but perhaps we _should_ check that they are coprime?

Reply:
At least we could check that neither is `0`.

Reply:
In practice this exchange rate will only be supplied by the node in serialized form. So we do not need to do many checks since the node will provide valid values.

Reply:
Yes, that does sound a bit surprising for the zero check. But ok, if we only ever get it from the node in practice, then it's fine.